### DISCUSSION TITLE
Flash builder is Asking for serial number

### DISCUSSION THREAD

**Discussion by ukmunnithan** [airsdk/Adobe-Runtime-Support#1390] [General]
We have taken licence and installed adobe flash builder 4.7 from cloud and it's not providing any licence key and installed & working.

But last few days, when open flash builder, it showing that its a trial licence and when signing to the interface it asks for serial number

Don't know why dialog is comes even having licence?

**Comment by ajwfrost** [comment]
I know that Flash Builder was originally a stand-alone product but then got merged into the Creative Cloud suite. So it may be that it's coming up with that dialog because it hasn't detected the Creative Cloud app or maybe Creative Cloud can't confirm the license validity. Worth ensuring that the Creative Cloud app says that everything is in order?

Flash Builder isn't officially supported any more though I'm afraid.. so you can't buy licenses for it specifically any more from Adobe, I don't actually know what the situation is with existing licenses if you then stop the Creative Cloud subscription..

thanks

**Comment by raresn** [comment]
I have spent hours with adobe support about this.
The only conclusion is that there isn't anything that can be done.
I am not saying that you should use a crack to be able to use their obsolete software, but as of now they don't offer a legit solution we can apply.

**Comment by 2jfw** [comment]
I have had this issue too and one workaround (found in Adobe Forums for Fireworks) is to install **Adobe MUSE** and sign-in to Adobe Cloud from there. It somehow will cause other legacy apps to resume working (e.g. Fireworks) - sounds strange but it reliably worked well so far

Also, have a look at the Adobe Fireworks forum to see the same problem for this app with different approaches to resolve:
https://community.adobe.com/t5/fireworks/ct-p/ct-fireworks?page=1&sort=latest_replies&filter=all&tabid=discussions

However, for me, only the trick with installing and starting/signing-in with MUSE worked

**Reply by barisyild** [reply]
Worked!

**Comment by hardcoremore** [comment]
Hi Guys,

I was somehow still able to use Flash Builder Premium 4.7 until today. Today it asked me to enter serial number and it would not let me in through Adobe Creative Cloud. From the link @2jfw posted I do not understand how to use Adobe MUSE to solve this.

If someone has serial number or crack I would be so thankful and gladly use it because all my projects are setup in Flash Builder which was working great for me and I could export release builds for PC until yesterday just fine. All my projects are setup in a complex way with Flash Builder where I have library projects and many more things so JUST using other IDE is not something that might work for me.

So PLEASE HELP IF YOU CAN. I can share my email or other info if someone can provide working serial number.

**Reply by 2jfw** [reply]
Assuming you have a valid Adobe Creative Cloud license, you will need to:
- Install *Muse* from the Adobe Creative Cloud
- Start Muse
- Muse will show a dialog and ask you to sing-in into Adobe Creative Cloud
- After singing-in from Muse application, you can start Flash Builder (/or Fireworks, which has the same problem) and it will no longer ask you to provide a key

**Reply by hardcoremore** [reply]
Hi @2jfw 

Thanks a lot for quick response. I have already tried that and went through Muse login dialog. It seams that they have been updated creative cloud to only allow starting trial for deprecated software. Even so I tried starting trial for muse which was only 7 days but without success. It even won't open muse and I still can not open Flash Builder 4.7

**Reply by 2jfw** [reply]
It could be that you need to sign-off of the creative cloud before you do some steps of the procedure.
Not sure when exactly but have a look at the "Correct" answer from the Adobe forums post.
It's been some time when I experienced the issue - as far as I remember it could be, that you need to be logged out of adobe cloud until Muse's or FlashBuilder's license/key dialog appears "correctly". I remember vaguely also having had a trial dialog but was able to resolve it with signing off I think ..

**Reply by hardcoremore** [reply]
I tried uninstalling Creative Cloud and installing Flash Builder again but its the same issue. I don't know what to do. My heart just sank.

How could I ever find serial number or crack for this.

Do you use Flash Builder today still?

**Reply by 2jfw** [reply]
I switched to IntelliJ a couple years ago and it's one of the best choices I made - no matter how complex and nested your project structure is, switching your IDE would be totally worth it (as FlashBuilder is quite the relic from the past...)

Did you just sign-out of Creative Cloud, then starting Muse, passing in your adobe account data in Muse, then starting Adobe Cloud (I think), and then FlashBuilder? Pretty sure the sign-off is required

**Reply by hardcoremore** [reply]
Yeah I tried sign off no luck.

I have one question regarding IntelliJ if you can answer it. The biggest reason I am using Flash Builder is because I have some library projects that are connected to main projects. For example I have this:

AdobeAIRBase ---> Library Project
Box2dBase ---> Library Project
ShapesInMotion ---> Library Project


ShapesInMotionAndroid ---> Mobile project for Android
ShapesInMotionPC ---> Adobe AIR Project for PC
ShapesInMotionIphone ---> Mobile project for Iphone

All these main projects are using all of the library projects above. So in every main project I am importing all 3 library projects

Could I replicate this inside IntelliJ?

Can you tell me please how IntelliJ supports Adobe AIR? Can I run debug versions from IDE itself or I need to use CMD for everything and INtelliJ just for code.

Thanks for all the help

**Reply by 2jfw** [reply]
> Could I replicate this inside IntelliJ?

Definitely - its not a big thing to set up the projects and connect or reference them.
The AIR support in IntelliJ is quite good I think and it's still maintained (unfortunately no direct AAB export for Android though).
Other than that, I'm not missing anything in IntelliJ.
There is a free trial version, just give it a try 👍

**Reply by hardcoremore** [reply]
Thanks for the reply. Can you remind me please what is direct AAB export for Android and how am I affected by this?

**Reply by 2jfw** [reply]
> Can I run debug versions from IDE itself or I need to use CMD for everything and INtelliJ just for code.

You can run debug like you are used to it from FB.
I would recommend you to download the 30 days trial and just setup a small project and compile it.
It's quite intuitive to set up and you will find many similarities to FB, so it should not be too hard for you.

Regarding AAB:
This is the "new" recommend output format for publishing Android apps.
Have a look here: https://airsdk.dev/docs/tutorials/platform/android/packaging-android-app-bundles
